src/worker/cloudfront.rs

```diff
@@ -49,30 +58,61 @@ impl CloudFront {
+                Err(error) => return OperationResult::Err(error),
+            };
+
+            let request = SignableRequest::from(&request);
+            let params = SigningParams::builder()
+                .access_key(&self.access_key)
+                .secret_key(&self.secret_key)
+                .region("us-east-1") // cloudfront is a regionless service, use the default region for signing.
+                .service_name("cloudfront")
+                .settings(SigningSettings::default())
+                .time(SystemTime::now())
+                .build()
+                .unwrap(); // all required fields are set
+
+            let (mut signature_headers, _) =
+                http_request::sign(request, &params).unwrap().into_parts();
 
-        let request = http::Request::post(&url).body(&body)?;
-        let request = SignableRequest::from(&request);
-        let params = SigningParams::builder()
-            .access_key(&self.access_key)
-            .secret_key(&self.secret_key)
-            .region("us-east-1") // cloudfront is a regionless service, use the default region for signing.
-            .service_name("cloudfront")
-            .settings(SigningSettings::default())
-            .time(SystemTime::now())
-            .build()
-            .unwrap(); // all required fields are set
-        let (mut signature_headers, _) = http_request::sign(request, &params).unwrap().into_parts();
-        let response = client
-            .post(url)
-            .headers(signature_headers.take_headers().unwrap_or_default())
-            .body(body)
-            .send()?;
-        if let Some(err) = response.error_for_status_ref().err() {
-            let headers = format!("{:?}", response.headers());
-            let body = response.text()?;
-            return Err(err).context(format!("{headers}; body: {body}"));
-        }
-        Ok(())
+            let response = match client
+                .post(&url)
+                .headers(signature_headers.take_headers().unwrap_or_default())
+                .body(body)
+                .send()
+                .context("Failed to send invalidation request")
+            {
+                Ok(response) => response,
+                Err(error) => return OperationResult::Retry(error),
+            };
+
+            let status = response.status();
+
+            let result = match response.error_for_status_ref() {
+                Ok(_) => Ok(()),
+                Err(error) => {
+                    let headers = format!("{:?}", response.headers());
+                    let body = response
+                        .text()
+                        .unwrap_or_else(|_| "Failed to decode response payload".to_string());
+
+                    Err(error).context(format!("{headers}; body: {body}"))
+                }
+            };
+
+            match result {
+                Ok(_) => OperationResult::Ok(()),
+                Err(error) if status.is_server_error() => OperationResult::Retry(error),
```

Comment:
We may want to retry on more than just server errors (5xx). The [AWS SDK](https://docs.aws.amazon.com/sdkref/latest/guide/feature-retry-behavior.html) says

> Transient errors (HTTP status codes 400, 408, 500, 502, 503, and 504) and throttling errors (HTTP status codes 400, 403, 429, 502, 503, and 509) can all be retried.

Reply:
hmm... it feels a bit weird to me to retry client errors, because those would indicate that something is wrong on our side 🤔 

I'd say let's try it with the current config, and hopefully the official AWS SDK will be stable soonish and then we can migrate to that.